client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -213,6 +216,17 @@ const UnifiedDesignPickerStep: StepType< {
 		number: 1000,
 	} );
 
+	/**
+	 * Load data needed for the ThemeTierBadge.
+	 *
+	 * TODO: Move this within ThemeTierBadge as it's the consumer of the data.
+	 * We will need to dedupe requests because right now each ThemeTierBadge
+	 * instance will trigger a new request.
+	 */
+	useQueryProductsList();
+	useQuerySiteFeatures( [ site?.ID ] );
+	useQuerySitePurchases( site?.ID ?? -1 );
```

Comment:
I created a [follow-up issue for this here](https://github.com/Automattic/wp-calypso/issues/102540).